archunit/src/main/java/com/tngtech/archunit/ArchConfiguration.java

```diff
@@ -273,10 +273,12 @@ public String getPropertyOrDefault(String propertyName, String defaultValue) {
                 ENABLE_MD5_IN_CLASS_SOURCES, Boolean.FALSE.toString()
         ));
 
-        private final Properties properties = createProperties(PROPERTY_DEFAULTS);
+        private Properties properties = createProperties(PROPERTY_DEFAULTS);
```

Comment:
Why did you make these non-`final`? In the original version the `Properties properties` never changed anyway, but only the `overwrittenProperties`, right?

Reply:
I did this because I misunderstood the structure of the code. After your explanation in one of these comments I see it does indeed not make sense